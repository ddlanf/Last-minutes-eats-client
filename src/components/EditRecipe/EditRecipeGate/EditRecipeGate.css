.edit-recipe-gate-form{ 
    margin-left: auto; 
    margin-right: auto; 
    display: block; 
    position: fixed;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    height:100%;
    padding-top: 20%;
    top: -2000px;
    left: 0;
    z-index:1000;
    background-color: rgba(0, 45, 82, 0.616);
    text-align: center; 
    animation-name: slide-down;
    animation-fill-mode: forwards;
    animation-duration: 0.5s;
    animation-timing-function: ease-in;
}

.slide-up{
    animation-name: slide-up;
    animation-duration: 0.5s;
    animation-fill-mode: forwards;
    animation-timing-function: linear;
}

@keyframes slide-down{
    0% { top: -2000px}
    90%{ top: 0}
    92%{ top: 30px }
    94%{ top: 0px }
    96%{ top: 20px }
    98%{ top: 0}
    99%{ top: 10px }
    100%{ top: 0}
}

@keyframes slide-up{
    0% { top: 0}
    100%{ top: -2000px; display: none; }
}

.edit-recipe-gate-label{
    font-size: 7vw;
    margin-bottom: 2vw;
    display: block;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    color: white;
}

.edit-recipe-gate-input{
    font-size: 6vw;
    width: 50%;
    max-width: 800px;
}

.edit-recipe-gate-submit{
    cursor: pointer;
    display: block;
    margin-left: auto;
    margin-right: auto;
    background-color: #209463;
    border: 1.5px solid #209463;
    color: white;
    border-radius: 3rem;
    padding: .5rem 2rem;
    font-size: 6vw;
    margin-top: 5%;
    width: 40%;
    max-width: 350px;
}

a.edit-recipe-gate-back{
    cursor: pointer;
    display: block;
    text-decoration: none;
    margin-left: auto;
    margin-right: auto;
    background-color: rgb(99, 99, 99);
    border: 1.5px solid rgb(99, 99, 99);
    color: white;
    border-radius: 3rem;
    padding: 1%;
    font-size: 6vw;
    margin-top: 5%;
    width: 32%;
    max-width: 300px;
}


.edit-recipe-gate-error{
    color: red;
    font-size: 5vw;
    background-color: #002D52;
    width: 50%;
    border-radius: 30%;
    margin-right: auto;
    margin-left: auto;
    margin-top: 2%;
}

.edit-recipe-gate-submitted{
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    font-size: 4.2vw;
}

@media screen and (min-width: 550px){
    .edit-recipe-gate-form{
       padding-top: 3%;
    }
}


@media screen and (min-width: 1000px){

    .edit-recipe-gate-form{
        padding-top: 3%;
     }

    .edit-recipe-gate-label{
        font-size: 4vw;
        margin-bottom: 5%;
    }

    .edit-recipe-gate-input{
        font-size: 4vw;
        width: 40%;
    }

    .edit-recipe-gate-submit{
        font-size: 50px;
    }

    a.edit-recipe-gate-back{
        font-size: 50px;
        padding: 0.5%;
    }

    .edit-recipe-gate-error{
        font-size: 3vw;
    }

    .edit-recipe-gate-submitted{
        font-size:  3vw;
        width: 100%;
    }

}