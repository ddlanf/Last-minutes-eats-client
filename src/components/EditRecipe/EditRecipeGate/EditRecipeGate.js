import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import RecipeTokenApiService from '../../../services/recipe-token-api-service'
import './EditRecipeGate.css'
export default class EditRecipeGate extends Component {

    state  = {
        error: '',
        changed: false,
        tokenAccepted: false,
        token: ''
    }

    handleInputChange = (event) =>{
        const { name } = event.target
        const { value } = event.target
        
        this.setState({
            [name] : value,
            changed : true,
        })
    }

    submitToken = (e) =>{
      e.preventDefault()

      RecipeTokenApiService.checkRecipeToken(this.state.token, this.props.recipeId)
            .then(res =>{
                this.props.setToken(this.state.token)
                this.setState({ tokenAccepted : true })
            })
            .catch(res =>{
                this.setState({ error: res.error })
            })
    }
    
    render() {
        return (
            <section className="edit-recipe-gate">
                    <form
                        onSubmit={(e) => this.submitToken(e)}
                        className={`edit-recipe-gate-form ${this.state.tokenAccepted ? 'slide-up':''}`}>
                        <label className="edit-recipe-gate-label">
                            Enter token to edit this recipe
                        </label>
                        <input 
                            type="password"
                            name="token"
                            onChange={this.handleInputChange}
                            className="edit-recipe-gate-input"/>
                        {this.state.error  ? <p className="edit-recipe-gate-error">{this.state.error}</p> : ''}
                        <button
                            disabled={!this.state.changed}
                            type="submit"
                            className="edit-recipe-gate-submit"
                            >
                            Submit
                        </button>
                        <Link
                            to={`/view-recipe/${this.props.recipeId}`}
                            className="edit-recipe-gate-back"
                            >
                            Back
                        </Link>
                    </form>
             </section>
        )
    }
}
