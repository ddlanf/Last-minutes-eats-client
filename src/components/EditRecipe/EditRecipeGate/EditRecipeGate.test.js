import React from 'react';
import ReactDOM from 'react-dom';
import EditRecipeGate from './EditRecipeGate';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('EditRecipeGate component', () => {

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>          
                        <EditRecipeGate/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<BrowserRouter><EditRecipeGate/></BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});