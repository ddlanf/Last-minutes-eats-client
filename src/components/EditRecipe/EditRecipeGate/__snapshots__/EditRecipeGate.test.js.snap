// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipeGate component renders the UI as expected 1`] = `
<section
  className="edit-recipe-gate"
>
  <form
    className="edit-recipe-gate-form "
    onSubmit={[Function]}
  >
    <label
      className="edit-recipe-gate-label"
    >
      Enter token to edit this recipe
    </label>
    <input
      className="edit-recipe-gate-input"
      name="token"
      onChange={[Function]}
      type="password"
    />
    
    <button
      className="edit-recipe-gate-submit"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
    <a
      className="edit-recipe-gate-back"
      href="/view-recipe/undefined"
      onClick={[Function]}
    >
      Back
    </a>
  </form>
</section>
`;
